src/components/shared/modals/modalWithInput/index.tsx

```diff
@@ -0,0 +1,122 @@
+import React, {useEffect} from 'react';
+import {Button, Modal, Text} from 'native-base';
+import Input from 'components/shared/input';
+import styles from './styles';
+import {useForm} from 'react-hook-form';
+import {schema} from './validations';
+import {yupResolver} from '@hookform/resolvers/yup';
+import {Keyboard} from 'react-native';
+
+interface Props {
+  buttonText: string;
+  headerText: string;
+  firstInputName: string;
+  secondInputName?: string;
+  onSubmit: () => void;
+}
+
+const ModalWithInput = ({
+  buttonText,
+  headerText,
+  firstInputName,
+  secondInputName,
+  onSubmit,
+}: Props) => {
+  const newObject = {
+    [firstInputName]: '',
+    [secondInputName]: '',
+  };
+
+  const [modalVisible, setModalVisible] = React.useState(false);
+  const {
+    control,
+    formState: {errors},
+    setError,
+    watch,
+    clearErrors,
+    handleSubmit,
+    setValue,
+  } = useForm({
+    defaultValues: newObject,
+    resolver: yupResolver(schema),
+    mode: 'onSubmit',
+    reValidateMode: 'onBlur',
+  });
+  const firstInputValue = watch(firstInputName);
+  const secondInputValue = watch(secondInputName);
+  const onPress = () => {
+    if (!firstInputValue || !secondInputValue) {
+      Keyboard.dismiss();
+      setError(firstInputName, {
+        type: 'Invalid email or password',
+        message: 'You need to enter a valid user and password',
+      });
+    }
+    onSubmit();
+  };
```

Comment:
The form data is injected by the handleSubmit(...) method which calls this function (on line 99). Therefore, we don't need to watch the fields to get their values. Also, the external onSubmit we are calling here should be aware of the form values entered.

We can update this piece code as follows, 
```suggestion


  const onPress = (data: {[x: string]: string}) => {
    if (!data.firstInputValue || !data.secondInputValue) {
      Keyboard.dismiss();
      setError(firstInputName, {
        type: 'Invalid email or password',
        message: 'You need to enter a valid user and password',
      });
    }
    onSubmit(data);
  };
```

Reply:
nice, didn't even think on that functionality of handlesubmit!, smart code review Martín!